apps/rmh-web/app/pt-portal/hooks.ts

```diff
@@ -0,0 +1,39 @@
+"use client";
+
+import "@aws-amplify/ui-react/styles.css";
+import "../globals.css";
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { User } from "./models";
+
+export const useUsers = (ptid: string) => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const loadAppointments = async () => {
+      try {
+        setIsLoading(true);
+        const { data }: { data: User[] } = await axios.get(
+          `http://localhost:3000/api/get-appointments?ptdid=${ptid}`
+        );
+
+        const sortedData = data.sort(
+          (a, b) =>
+            new Date(a.appointmentDate).getTime() -
+            new Date(b.appointmentDate).getTime()
+        );
+
+        setUsers(sortedData);
+      } catch (err) {
+        throw err;
```

Comment:
we could probably use [userSwr](https://swr.vercel.app/) here  or something, but for the sake of this task I used axios.  Also probably want to handle the various errors that be returned instead of throwing it